xml/value.go

```diff
@@ -0,0 +1,339 @@
+package xml
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math"
+	"math/big"
+	"strconv"
+	"strings"
+)
+
+// Value represents an XML Value type
+// XML Value types: Object, Array, Map, String, Number, Boolean, and Null.
+type Value struct {
+	w       writer
+	scratch *[]byte
+
+	startElement *StartElement
+	endElement   *EndElement
+}
+
+// newValue returns a new Value encoder
+func newValue(w writer, scratch *[]byte, startElement *StartElement, endElement *EndElement) Value {
+	return Value{
+		w:            w,
+		scratch:      scratch,
+		startElement: startElement,
+		endElement:   endElement,
+	}
+}
+
+// writeStartElement takes in a start element and writes it.
+// It handles namespace, attributes in start element.
+func writeStartElement(w writer, el *StartElement) error {
+	if el == nil {
+		return nil
+	}
+
+	w.WriteRune(leftAngleBracket)
+
+	if len(el.Name.Space) != 0 {
+		w.WriteString(el.Name.Space)
+		w.WriteRune(colon)
+	}
+	w.WriteString(el.Name.Local)
+
+	for _, attr := range el.Attr {
+		// if attribute name len is zero or name.space is `xmlns`, it is a namespace attribute
```

Comment:
Nit the comment is a repeat of the if statement. Maybe give clarrification why we treat a empty name attribute as a namespace?